### PR TITLE
Add a service protocol for raster cache memory

### PR DESCRIPTION
**Pull request by liyuqian**
Related issue: https://github.com/flutter/flutter/issues/56719

### CODE REVIEW

File: shell/common/shell.cc [hunk 1]
```diff
@@ -1424,6 +1429,23 @@ bool Shell::OnServiceProtocolGetSkSLs(
...
 
+bool Shell::OnServiceProtocolEstimateRasterCacheMemory(
+    const ServiceProtocol::Handler::ServiceProtocolMap& params,
+    rapidjson::Document* response) {
+  FML_DCHECK(task_runners_.GetRasterTaskRunner()->RunsTasksOnCurrentThread());
+  const auto& raster_cache = rasterizer_->compositor_context()->raster_cache();
+  response->SetObject();
+  response->AddMember("type", "EstimateRasterCacheMemory",
+                      response->GetAllocator());
+  response->AddMember<uint64_t>("layer_bytes",
```
Comment (jacob314) [inline shell/common/shell.cc:1440]: nit:  
can you call these `layerBytes` and `pictureBytes` or `layerSizeInBytes` and `pictureSizeInBytes` for consistency with other apis?
using camel case makes the API a little cleaner to use from Dart.

File: flow/raster_cache.cc [hunk 2]
```diff
@@ -299,35 +299,33 @@ void RasterCache::SetCheckboardCacheImages(bool checkerboard) {
 
 void RasterCache::TraceStatsToTimeline() const {
 #if !FLUTTER_RELEASE
+  FML_TRACE_COUNTER("flutter", "RasterCache", reinterpret_cast<int64_t>(this),
+                    "LayerCount", layer_cache_.size(), "LayerMBytes",
+                    EstimateLayerCacheByteSize() * 1e-6, "PictureCount",
```
Comment (dnfield) [inline flow/raster_cache.cc:304]: nit: we might want to consider making a constexpr like `kMegaBytes` or something for this. I'm thinking the same for another PR where I'm dividing by 1,000,000 to the same effect.

File: shell/common/shell.cc (hunk 1 above)
Comment (liyuqian) [reply shell/common/shell.cc]: Changed to `layerBytes` and `pictureBytes`.

File: flow/raster_cache.cc (hunk 2 above)
Comment (dnfield) [reply flow/raster_cache.cc]: Can we make it be 2^20 to match the way dart vm calculates mbytes?

Comment (liyuqian) [reply flow/raster_cache.cc]: Sure, changed it to `1 << 20`. Does it look good?

**Pull request merged**